net.c

```diff
@@ -236,7 +236,7 @@ int redisContextSetTcpUserTimeout(redisContext *c, unsigned int timeout) {
     res = -1;
     (void)timeout;
 #endif
-    if (res == -1); {
+    if (res == -1) {
         __redisSetErrorFromErrno(c,REDIS_ERR_IO,"setsockopt(TCP_USER_TIMEOUT)");
         redisNetClose(c);
         return REDIS_ERR;
```

Comment:
What! That's a disaster. :) It would always return REDIS_ERR.

I should have used this in a test. It would have caught it.

Sorry for this and well spotted @yossigo.